include/kernel.h

```diff
@@ -3030,10 +3030,12 @@ __syscall int k_mutex_lock(struct k_mutex *mutex, s32_t timeout);
  *
  * @param mutex Address of the mutex.
  *
- * @return N/A
+ * @retval 0 Mutex unlocked.
+ * @retval -EPERM The current thread does not own the mutex.
+ * @retval -EINVAL Undefined behavior.
```

Comment:
EINVAL is for invalid parameter not undefined behavior

Reply:
btw, return EINVAL seems fine, can't we just say that mutex is not locked

Reply:
I am going to follow posix here and return EPERM in both cases